File: Models/Vehicle.cs
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Vegas.Models
{
    [Table("Vehicles")]
    public class Vehicle
    {
        public Vehicle() 
        {
            this.VehicleFeatures = new Collection<VehicleFeature>();
        }
        public int Id { get; set; }
        public Make Make { get; set; }
        public int MakeId { get; set; }
        [Required]
        public Model Model { get; set; }
        public int ModelId { get; set; }
        public bool IsRegistered { get; set; }
        [Required]
        [MaxLength(255)]
        public string ContactName  { get; set; }
        [Required]
        [MaxLength(255)]
        public string ContactPhone { get; set; }
        [MaxLength(255)]
        public string ContactEmail { get; set; }
        public ICollection<VehicleFeature> VehicleFeatures { get; set; }
    }
}

File: ClientApp/src/app/models/Vehicle.ts
export interface KeyValuePair{
    id: number;
    name: string;
}

export interface Contact{
    name: string;
    phone: string;
    email: string;
}

export interface Vehicle {
    id: number;
    model: KeyValuePair;
    make: KeyValuePair;
    isRegistered: boolean;
    contact: Contact;
    features: KeyValuePair[];
    lastUpdate: string;
}

export interface SaveVehicle{
    id: number;
    modelId: number;
    makeId: number;
    isRegistered: boolean;
    contact: Contact;
    features: number[];
}